jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/CombinedResource.java

```diff
@@ -26,36 +23,52 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.eclipse.jetty.util.URIUtil;
 
 /**
- * A collection of Resources.
- * Allows webapps to have multiple sources.
- * The first resource in the collection is the main resource.
- * If a resource is not found in the main resource, it looks it up in
- * the order the provided in the constructors
+ * Multiple resource directories presenteda as a single Resource.
  */
-public class ResourceCollection extends Resource
+public class CombinedResource extends Resource
 {
-    private final List<Resource> _resources;
+    /**
+     * Stream a resource to its component resources (if any).
+     * @param resource The resource to stream
+     * @return If the passed resource is a ResourceCollection, then it's contained
+     *         resources are streamed, otherwise the passed resource itself if streamed
+     */
+    public static Stream<Resource> stream(Resource resource)
+    {
+        if (resource instanceof CombinedResource rc)
+            return rc.getResources().stream();
+        return Stream.of(resource);
+    }
 
     /**
-     * Instantiates a new resource collection.
-     *
-     * @param resources the resources to be added to collection
+     * <p>Make a Resource containing a combination of other resources</p>
+     * @param resources multiple resources to combine as a single resource. Typically, they are directories.
+     * @return A Resource of multiple resources or a single resource if only 1 is passed, or null if none are passed
+     * @see CombinedResource
      */
-    ResourceCollection(List<Resource> resources)
+    static Resource combine(List<Resource> resources)
```

Comment:
Isn't this on `ResourceFactory.combine()` already?

Reply:
It is, which calls this version, which has access to the private `gatherUniqueFlatResourceList` method which then calls the private constructor, that no longer does the sanity checks, as they are not needed when creating a CombinedResource from resolve.   So this allows us to avoid needless iterations and checks.

Reply:
Note that this method  is only package scope, so it is not public API.  It is just to get the implementation in the right file rather than in ResourceFactory.java